# Project 1: Deaths by tuberculosis

by Zakia Syeda, 8 December 2017

This is the project notebook for Week 1 of The Open University's [_Learn to code for Data Analysis_](http://futurelearn.com/courses/learn-to-code) course.

In 2000, the United Nations set eight Millenium Development Goals (MDGs) to reduce poverty and diseases, improve gender equality and environmental sustainability, etc. Each goal is quantified and time-bound, to be achieved by the end of 2015. Goal 6 is to have halted and started reversing the spread of HIV, malaria and tuberculosis (TB).
TB doesn't make headlines like Ebola, SARS (severe acute respiratory syndrome) and other epidemics, but is far deadlier. For more information, see the World Health Organisation (WHO) page <http://www.who.int/gho/tb/en/>.

Given the population and number of deaths due to TB in some countries during one year, the following questions will be answered: 

- What is the total, maximum, minimum and average number of deaths in that year?
- Which countries have the most and the least deaths?
- What is the death rate (deaths per 100,000 inhabitants) for each country?
- Which countries have the lowest and highest death rate?

The death rate allows for a better comparison of countries with widely different population sizes.

## The data

The data consists of total population and total number of deaths due to TB (excluding HIV) in 2015 in each of the Common Wealth Countries. 

The data was taken in December 2017 from <http://apps.who.int/gho/data/node.main.POP107?lang=en> (population) and <http://apps.who.int/gho/data/node.main.593?lang=en> (deaths). The uncertainty bounds of the number of deaths were ignored.

The data was collected into an Excel file which should be in the same folder as this notebook.

In [2]:
import warnings
warnings.simplefilter('ignore', FutureWarning)

from pandas import *
data = read_excel('Who pop tb Common Wealth Countries 2015.xlsx')
data

Country  Population (1000s)  TB deaths
0           Antigua and Barbuda                91.8        0.0
1                     Australia             23969.0       47.0
2                       Bahamas               388.0        1.0
3                    Bangladesh            160996.0    70000.0
4                      Barbados               284.2        2.0
5                        Belize               359.3        8.0
6                      Botswana              2262.5      560.0
7             Brunei Darussalam               423.2        2.3
8                      Cameroon            233442.0     6600.0
9                        Canada             35939.9       98.0
10                       Cyprus              1165.3        5.0
11                     Dominica                72.7        0.0
12                         Fiji               892.1       38.0
13                        Ghana             27409.9    10000.0
14                      Grenada               106.8        2.0
15                       Guyana               767.1      140.0
16                        India           1252140.0   434000.0
17                      Jamaica              2793.3        7.0
18                        Kenya             46050.3    30000.0
19                     Kiribati               112.4       36.0
20                      Lesotho              2135.0     1200.0
21                       Malawi             17215.2     4000.0
22                     Malaysia             30331.0     1300.0
23                        Malta               418.7        1.0
24                    Mauritius              1273.2       20.0
25                   Mozambique             27977.9    22000.0
26                      Namibia              2458.8      790.0
27                        Nauru                10.2        1.0
28                  New Zealand              4528.5       10.0
29                      Nigeria            182202.0   112000.0
30                     Pakistan            188925.0    45000.0
31             Papua New Guinea             76193.0     3700.0
32                       Rwanda             11609.7      460.0
33        Saint Kitts and Nevis                55.6        1.0
34                  Saint Lucia               185.0        4.0
35                Saint Vincent               109.5        1.0
36                        Samoa               193.2        2.0
37                   Seychelles                96.5        0.0
38                 Sierra Leone              6453.2     4100.0
39                    Singapore              5603.7       53.0
40              Solomon Islands               583.6       57.0
41                 South Africa             54490.4    23000.0
42                    Sri Lanka             20715.0     1200.0
43                    Swaziland              1287.0      290.0
44                        Tonga               106.2        1.0
45          Trinidad and Tobogo              1360.1       23.0
46                       Tuvalu                 9.9        2.0
47                       Uganda             39032.4    10000.0
48               United Kingdom             64715.8      350.0
49  United Republic of Tanzania             53470.4    30000.0
50                      Vanuatu               264.7       17.0
51                       Zambia             16211.8     5000.0
52                     Zimbabwe             15602.8     1800.0

## The range of the problem

The column of interest is the last one.

In [3]:
tbColumn = data['TB deaths']

The total number of deaths in 2015 is:

In [4]:
tbColumn.sum()

817929.3

The largest and smallest number of deaths in a single country are:

In [5]:
tbColumn.max()

434000.0

In [6]:
tbColumn.min()

0.0

From less than 1 to almost a quarter of a million deaths is a huge range. The average number of deaths, over all countries in the data, can give a better idea of the seriousness of the problem in each country.
The average can be computed as the mean or the median. Given the wide range of deaths, the median is probably a more sensible average measure.

In [7]:
tbColumn.mean()

15432.628301886793

In [8]:
tbColumn.median()

57.0

The median is far lower than the mean. This indicates that some of the countries had a very high number of TB deaths in 2015, pushing the value of the mean up.

## The most affected

To see the most affected countries, the table is sorted in ascending order by the last column, which puts those countries in the last rows.

In [9]:
data.sort_values ('TB deaths')

Country  Population (1000s)  TB deaths
0           Antigua and Barbuda                91.8        0.0
37                   Seychelles                96.5        0.0
11                     Dominica                72.7        0.0
44                        Tonga               106.2        1.0
33        Saint Kitts and Nevis                55.6        1.0
27                        Nauru                10.2        1.0
23                        Malta               418.7        1.0
2                       Bahamas               388.0        1.0
35                Saint Vincent               109.5        1.0
4                      Barbados               284.2        2.0
36                        Samoa               193.2        2.0
46                       Tuvalu                 9.9        2.0
14                      Grenada               106.8        2.0
7             Brunei Darussalam               423.2        2.3
34                  Saint Lucia               185.0        4.0
10                       Cyprus              1165.3        5.0
17                      Jamaica              2793.3        7.0
5                        Belize               359.3        8.0
28                  New Zealand              4528.5       10.0
50                      Vanuatu               264.7       17.0
24                    Mauritius              1273.2       20.0
45          Trinidad and Tobogo              1360.1       23.0
19                     Kiribati               112.4       36.0
12                         Fiji               892.1       38.0
1                     Australia             23969.0       47.0
39                    Singapore              5603.7       53.0
40              Solomon Islands               583.6       57.0
9                        Canada             35939.9       98.0
15                       Guyana               767.1      140.0
43                    Swaziland              1287.0      290.0
48               United Kingdom             64715.8      350.0
32                       Rwanda             11609.7      460.0
6                      Botswana              2262.5      560.0
26                      Namibia              2458.8      790.0
20                      Lesotho              2135.0     1200.0
42                    Sri Lanka             20715.0     1200.0
22                     Malaysia             30331.0     1300.0
52                     Zimbabwe             15602.8     1800.0
31             Papua New Guinea             76193.0     3700.0
21                       Malawi             17215.2     4000.0
38                 Sierra Leone              6453.2     4100.0
51                       Zambia             16211.8     5000.0
8                      Cameroon            233442.0     6600.0
13                        Ghana             27409.9    10000.0
47                       Uganda             39032.4    10000.0
25                   Mozambique             27977.9    22000.0
41                 South Africa             54490.4    23000.0
18                        Kenya             46050.3    30000.0
49  United Republic of Tanzania             53470.4    30000.0
30                     Pakistan            188925.0    45000.0
3                    Bangladesh            160996.0    70000.0
29                      Nigeria            182202.0   112000.0
16                        India           1252140.0   434000.0

The table raises the possibility that a large number of deaths may be partly due to a large population. To compare the countries on an equal footing, the death rate per 100,000 inhabitants is computed.

In [10]:
populationColumn = data['Population (1000s)']
data['TB deaths (per 100,000)'] = tbColumn * 100 / populationColumn
data

Country  Population (1000s)  TB deaths  \
0           Antigua and Barbuda                91.8        0.0   
1                     Australia             23969.0       47.0   
2                       Bahamas               388.0        1.0   
3                    Bangladesh            160996.0    70000.0   
4                      Barbados               284.2        2.0   
5                        Belize               359.3        8.0   
6                      Botswana              2262.5      560.0   
7             Brunei Darussalam               423.2        2.3   
8                      Cameroon            233442.0     6600.0   
9                        Canada             35939.9       98.0   
10                       Cyprus              1165.3        5.0   
11                     Dominica                72.7        0.0   
12                         Fiji               892.1       38.0   
13                        Ghana             27409.9    10000.0   
14                      Grenada               106.8        2.0   
15                       Guyana               767.1      140.0   
16                        India           1252140.0   434000.0   
17                      Jamaica              2793.3        7.0   
18                        Kenya             46050.3    30000.0   
19                     Kiribati               112.4       36.0   
20                      Lesotho              2135.0     1200.0   
21                       Malawi             17215.2     4000.0   
22                     Malaysia             30331.0     1300.0   
23                        Malta               418.7        1.0   
24                    Mauritius              1273.2       20.0   
25                   Mozambique             27977.9    22000.0   
26                      Namibia              2458.8      790.0   
27                        Nauru                10.2        1.0   
28                  New Zealand              4528.5       10.0   
29                      Nigeria            182202.0   112000.0   
30                     Pakistan            188925.0    45000.0   
31             Papua New Guinea             76193.0     3700.0   
32                       Rwanda             11609.7      460.0   
33        Saint Kitts and Nevis                55.6        1.0   
34                  Saint Lucia               185.0        4.0   
35                Saint Vincent               109.5        1.0   
36                        Samoa               193.2        2.0   
37                   Seychelles                96.5        0.0   
38                 Sierra Leone              6453.2     4100.0   
39                    Singapore              5603.7       53.0   
40              Solomon Islands               583.6       57.0   
41                 South Africa             54490.4    23000.0   
42                    Sri Lanka             20715.0     1200.0   
43                    Swaziland              1287.0      290.0   
44                        Tonga               106.2        1.0   
45          Trinidad and Tobogo              1360.1       23.0   
46                       Tuvalu                 9.9        2.0   
47                       Uganda             39032.4    10000.0   
48               United Kingdom             64715.8      350.0   
49  United Republic of Tanzania             53470.4    30000.0   
50                      Vanuatu               264.7       17.0   
51                       Zambia             16211.8     5000.0   
52                     Zimbabwe             15602.8     1800.0   

    TB deaths (per 100,000)  
0                  0.000000  
1                  0.196087  
2                  0.257732  
3                 43.479341  
4                  0.703730  
5                  2.226552  
6                 24.751381  
7                  0.543478  
8                  2.827255  
9                  0.272677  
10                 0.429074  
11                 0.000000  
12                 4.259612  
13                36.483168  
14       

## Conclusions

The Common Wealth Countries had a total of about 817 thousand deaths due to TB in 2015. The median shows that half of these countries had fewer than 57 deaths. The much higher mean (over 15,000) indicates that some countries had a very high number. The least affected were Antigua and Barbuda, Seychelles and Dominica, with less than one death (as the values are estimates), and the most affected were India and Nigeria with 434 thousand and 112 thousand deaths in a single year. However, taking the population size into account, the least affected were Antigua and Barbuda and Dominica with less than 1 death per 100 thousand inhabitants, and the most affected were Kenya and Mozambique with over 60 and over 70 deaths per 100,000 inhabitants respectively.

One should not forget that most values are estimates, and that the chosen countries are a small sample of all the world's countries. Nevertheless, they convey the message that TB is still a major cause of fatalities, and that there is a huge disparity between countries, with several ones being highly affected.